include/sys/util.h

```diff
@@ -471,7 +471,7 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
  *
  * @retval expr As a boolean return, if false then it has timed out.
  */
-#define wait_for(expr, timeout, delay_stmt)                                                        \
```

Comment:
No where in our docs is it required that a macro be capitalized, and many kernel.h macros are not. Many kernel.h macros are also macro expressions. See k_fifo_get as one of many examples. The docs specifically say follow the linux kernel style. Linux kernel style docs: "CAPITALIZED macro names are appreciated but macros resembling functions may be named in lower case."

I'm in pretty strong disagreement that this change is needed.

Reply:
Because the macro uses the same name as a C++ library function, the C++ function gets expanded by the macro. Using explicit namespaces doesn't fix it.

We could fix this by editing the C++ standard library, or by renaming this macro. One of those is much easier (and faster) than the other.

Reply:
Well there's no fixing the C++ standard library, so lets rename this instead, I agree. Makes sense.